tests/test_decoding.py

```diff
@@ -25,7 +21,37 @@ class SomeClass:
     assert pyrallis.decode(SomeClass, pyrallis.encode(b)) == b
 
 
-def test_dump_load(simple_attribute, tmp_path):
+@parametrize('config_type', ['', 'yaml', 'json', 'toml'])
+def test_dump_load(simple_attribute, config_type, tmp_path):
+    some_type, passed_value, expected_value = simple_attribute
+
+    if config_type != '':
+        pyrallis.set_config_type(config_type)
+
+    @dataclass
+    class SomeClass:
+        val: Optional[some_type] = None
+
+    b = SomeClass()
+    b.val = expected_value
+
+    tmp_file = tmp_path / f'config'
+    pyrallis.dump(b, tmp_file.open('w'))
+
+    new_b = pyrallis.parse(config_class=SomeClass, config_path=tmp_file, args="")
+    assert new_b == b
+
+    arguments = shlex.split(f"--config_path {tmp_file}")
+    new_b = pyrallis.parse(config_class=SomeClass, args=arguments)
+    assert new_b == b
+
+    new_b = pyrallis.parse(config_class=SomeClass, args="")
+    assert new_b != b
+
+    pyrallis.set_config_type('yaml')
```

Comment:
I'm afraid that this is error prone. I think you should also have a context manager for config type so you can change it temporarily in a safe manner, i.e. w/o side-effects. I understand that this can lead to unwanted behavior (conflicts stemming from use of multiple formats in a single flow) but I think it's explicit enough to trust the user, and can be useful for some cases (e.g. here in this test)

See for example pandas' `pandas.option_context` or numpy's `numpy.printoptions` (which can be set globally too)

Reply:
Although in second thought, you may want to test both methods either way.

Reply:
Good point, need to give this some more thought